scan-command/src/main/java/io/ballerina/scan/internal/StaticCodeAnalyzer.java

```diff
@@ -72,6 +80,70 @@ public void visit(CheckExpressionNode checkExpressionNode) {
         }
     }
 
+    public void visit(BinaryExpressionNode binaryExpressionNode) {
+        if (binaryExpressionNode.operator().kind().equals(ELLIPSIS_TOKEN)
+                || binaryExpressionNode.operator().kind().equals(DOUBLE_DOT_LT_TOKEN)) {
+            validateRangeExpressionOperator(scannerContext, document, binaryExpressionNode.lhsExpr(),
+                    binaryExpressionNode.rhsExpr(), binaryExpressionNode.operator(), binaryExpressionNode.location());
+        }
+    }
+
+    private void validateRangeExpressionOperator(ScannerContext scannerContext, Document document,
+                                                 Node n1, Node n2, Token operator, NodeLocation location) {
+        String lhsExpr = null;
+        String rhsExpr = null;
+        boolean isMinusOperatorPresentInLhs = false;
+        boolean isMinusOperatorPresentInRhs = false;
+
+        if (n1 instanceof UnaryExpressionNode unaryN1) {
+            if (unaryN1.unaryOperator().kind() == SyntaxKind.MINUS_TOKEN) {
+                isMinusOperatorPresentInLhs = true;
+            }
+            n1 = unaryN1.expression();
+        }
+        if (n2 instanceof UnaryExpressionNode unaryN2) {
+            if (unaryN2.unaryOperator().kind() == SyntaxKind.MINUS_TOKEN) {
+                isMinusOperatorPresentInRhs = true;
+            }
+            n2 = unaryN2.expression();
+        }
+
+        if (n1 instanceof BasicLiteralNode lhsExprNode) {
+            lhsExpr = lhsExprNode.literalToken().text();
+            if (isMinusOperatorPresentInLhs) {
+                lhsExpr = "-" + lhsExpr;
+            }
+        }
+
+        if (n2 instanceof BasicLiteralNode rhsExprNode) {
+            rhsExpr = rhsExprNode.literalToken().text();
+            if (isMinusOperatorPresentInRhs) {
+                rhsExpr = "-" + rhsExpr;
+            }
+        }
+
+        if (lhsExpr != null && rhsExpr != null) {
+            // According to the spec, these literal tokens are integers. So no need to cast or check.
+            try {
+                int lhs = Integer.parseInt(lhsExpr);
+                int rhs = Integer.parseInt(rhsExpr);
+                if (operator.kind() == DOUBLE_DOT_LT_TOKEN) {
+                    if (lhs >= rhs) {
+                        scannerContext.getReporter().reportIssue(document, location,
+                                CoreRule.INVALID_RANGE_OPERATOR.rule());
+                    }
+                } else if (operator.kind() == ELLIPSIS_TOKEN) {
```

Comment:
Isn't this always true? Can be an else (can also use early returns then).

Reply:
I guess it is better to use else if and show the condition.
It will Improve the readability